client/dashboard/task-list/tasks/payments/stripe.js

```diff
@@ -0,0 +1,317 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { Component, Fragment } from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+import apiFetch from '@wordpress/api-fetch';
+import { withDispatch } from '@wordpress/data';
+import interpolateComponents from 'interpolate-components';
+import { Modal } from '@wordpress/components';
+import { Button, TextControl } from 'newspack-components';
+
+/**
+ * WooCommerce dependencies
+ */
+import { Form, Link } from '@woocommerce/components';
+import withSelect from 'wc-api/with-select';
+import { WCS_NAMESPACE } from 'wc-api/constants';
+
+class Stripe extends Component {
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			showErrorModal: false,
+			errorMessage: '',
+			connectURL: '',
+			showConnectionButtons: ! props.manualConfig && ! props.createAccount,
+			showManualConfiguration: props.manualConfig,
+		};
+
+		this.updateSettings = this.updateSettings.bind( this );
+	}
+
+	componentDidMount() {
+		const { createAccount } = this.props;
+		const { showConnectionButtons } = this.state;
+
+		if ( createAccount ) {
+			this.autoCreateAccount();
+		}
+
+		if ( showConnectionButtons ) {
+			this.fetchOAuthConnectURL();
+		}
+	}
+
+	componentDidUpdate( prevProps, prevState ) {
+		if ( false === prevState.showConnectionButtons && this.state.showConnectionButtons ) {
+			this.fetchOAuthConnectURL();
+		}
+	}
+
+	async fetchOAuthConnectURL() {
+		const { returnUrl } = this.props;
+		try {
+			const result = await apiFetch( {
+				path: WCS_NAMESPACE + '/connect/stripe/oauth/init',
+				method: 'POST',
+				data: {
+					returnUrl,
+				},
+			} );
+			if ( ! result || ! result.oauthUrl ) {
+				this.setState( {
+					showConnectionButtons: false,
+					showManualConfiguration: true,
+				} );
+				return;
+			}
+			this.setState( {
+				connectURL: result.oauthUrl,
+			} );
+		} catch ( error ) {
+			// Fallback to manual configuration if the OAuth URL cannot be grabbed.
+			this.setState( {
+				showConnectionButtons: false,
+				showManualConfiguration: true,
+			} );
+		}
+	}
+
+	async autoCreateAccount() {
+		const { email, countryCode } = this.props;
+		try {
+			const result = await apiFetch( {
+				path: WCS_NAMESPACE + '/connect/stripe/account',
```

Comment:
I'm getting a 404 here - `/wp-json/wc/v1/connect/stripe/account?_locale=user`

WCS is installed and Jetpack is connected.

Reply:
`WooCommerce Stripe Gateway` is correctly installed and activated, correct? The first steps should do this correctly, but I just want to make sure.

https://github.com/Automattic/woocommerce-services/blob/29dfe0ba6fd3075afe08f917a6ff33c321502d9c/woocommerce-services.php#L831-L847

https://github.com/Automattic/woocommerce-services/blob/29dfe0ba6fd3075afe08f917a6ff33c321502d9c/classes/class-wc-connect-stripe.php#L42-L44

Do you see this endpoint under `v2` or `v3` instead?

Reply:
Thanks for those links- traced it back to this line:

https://github.com/Automattic/woocommerce-services/blob/29dfe0ba6fd3075afe08f917a6ff33c321502d9c/woocommerce-services.php#L446

Seems like `$tos_accepted` is false.  Should we prompt users to accept somewhere or hide/disable this feature if not accepted?

Reply:
It looks like we can set `update_option( 'woocommerce_setup_jetpack_opted_in', true );` to true and the TOS will get auto-accepted.

I think we might need some additional language for this though. Asked in Slack. p1569351775011500-wc-onboarding-slack.

Reply:
Added a TOS message and acceptance to the profile wizard in https://github.com/woocommerce/woocommerce-admin/pull/2897/commits/5be65a516aff4a62e0647bc9e0d79d4b076f7a9e.